numpy/random/mtrand/distributions.c

```diff
@@ -500,6 +500,10 @@ long rk_poisson_mult(rk_state *state, double lam)
     }
 }
 
+/*
```

Comment:
thats another paper by the same author, likely generalized
the exact algorithm was first published in W. Hörmann 
The transformed rejection method for generating Poisson random variables Insurance: Mathematics and Economics 12, 39-45 (1993)
I kept it short as you can find the thing by googling the numbers involved so I though the title ie enough ;)